### ISSUE TITLE
[amp-animation] allow parametrization via action

### ISSUE THREAD

**Issue by sebastianbenz** [ampproject/amphtml#16014]
Currently, there's no way to apply the same animation to different elements. It'd be great if there'd be a way to parameterize an amp-animation when it's started. For example:

```
myAnimation.start({'selector', '#myElement'})
```

I see two options:

1. A whitelisted set of options, e.g. `selector` that gets applied to every animation that's missing a selector. 
2. Arbitrary JSON that gets merged into the existing amp-animation definition (similar to AMP.setState). 

Personally, I'd prefer option 2 as it seems the most flexible (and easiest to implement).

**Comment by aghassemi** [comment]
Another option is `myAnimation.setTarget()`.

**Comment by nainar** [comment]
I think separating the merging of JSON from executing an action is a good idea. 

The current design that @aghassemi and I discussed offline is the following:
`animation.mergeDefinition(JSON)`. Which on execution merges the provided JSON with the already provided definition and then:
 - If the animation is already playing: restarts it with the new config. 
 - If the animation is paused - restarts it with the new config when restarted via calls to `seekTo`, `togglePause`, `restart`, etc.
 - If the animation is stopped - the next run starts it with the new config.

This makes sure we don't have clashes in running the animation. 

Another question worth considering is do we only want to allow modifications to parameters in the animations or in the timing config (`duration`, `iterations`, `fill`, `direction` ) as well?

**Comment by sebastianbenz** [comment]
Is there is a strict 1:1 association between an element and an animation or could the same animation run in parallel with different targets? If it's a 1:1 association, `mergeDefinition` could work - otherwise not.

**Comment by aghassemi** [comment]
There can be multiple `animations` inside a single `amp-animation` targeting different elements. `mergeDefinition` can still work in that case, but it is not pretty, let's say you want to change the target of second animation but now first one:
`mergeDefinitions: [{}, {target: #new}]` would result in that when merged with the original definition.

What we need here is an API to change arbitrary JSON. Few options:
1- Customized API for every single JSON property, e.g. `getAnimation(2).setTarget(`newTarget`)`. This will create a huge API surface that needs to be updated as new properties are added.
2- Something like JSONPath ( https://github.com/json-path/JsonPath ) which is like an XPATH API but for JSON. e.g. myAnimation.animations[2].target = $2. Down side of this is the extra parsing logic that needs to be added to the JS file. But it may be acceptable. 
3- `merge` which is similar to what `amp-bind` ended up doing. It is not perfect specially when it comes to array access, but works and is consistent with `amp-bind`.

**Comment by nainar** [comment]
Won't option 2 here also add a dependency load? 

I prefer option 3 the most myself, understanding the caveat that it is a bit verbose to express.
